controller/cache/cluster.go

```diff
@@ -530,9 +530,12 @@ func (c *clusterInfo) processEvent(event watch.EventType, un *unstructured.Unstr
 	if c.onEventReceived != nil {
 		c.onEventReceived(event, un)
 	}
+	key := kube.GetResourceKey(un)
```

Comment:
We should skip only `watch.Modified` events. Otherwise, we might get stale endpoints in cache. The simplest way to reproduce it: try to delete EndPoint in the UI. Since we are ignoring EndPoint events completely UI will keep showing deleted EndPoint.

Reply:
@alexmt done